### PR TITLE
feat(console)!: require sign in

### PR DESCRIPTION
**Pull request by skyrpex**
Users are now required to sign in to the Wing Console using their GitHub account.

This allows us to connect with those using the Wing toolchain, enabling us to gather feedback on what they like, what could be improved, and more. We believe this will help us make more informed product decisions in the future.

Related to https://github.com/winglang/wing/issues/5517.

### CODE REVIEW

**Comment by eladb** [comment]
@skyrpex when you are ready, can you share a screen recording of the experience?

**Comment by skyrpex** [comment]
@eladb this is roughly the experience. It's slow because the simulator for Wing Cloud is slow on my end, and we're also missing the GitHub login page in the video because I already authenticated once in the past.

https://github.com/winglang/wing/assets/1077520/235ad00a-8610-46b5-8e42-6e5a1d2eccda

**Comment by skyrpex** [comment]
The PR is done but let's keep it as a draft until we don't merge the related Wing Cloud PR.

**Comment by skyrpex** [comment]
⚠️ The GitHub CI will need a new `VITE_WING_CLOUD_SIGN_IN_URL` environment variable, pointing to `https://<wingcloud>/wrpc/console.signIn`.

**Review by ainvoner — CHANGES_REQUESTED** [review]

**Review by ainvoner — APPROVED** [review]

File: apps/wing-console/console/server/src/router/app.ts
Comment (skyrpex) [reply apps/wing-console/console/server/src/router/app.ts:55]: We keep it for backwards compatibility, mostly because of possible version mismatches between Console frontend and Console backend. Not sure when or if it can happen, though.

Do you think we should just remove it?

Comment (ainvoner) [reply apps/wing-console/console/server/src/router/app.ts:55]: if for all current console usages there is no need for backward comp. - remove it.
Wing Cloud
Console (CLI)
Playground
VSCode

Comment (skyrpex) [reply apps/wing-console/console/server/src/router/app.ts]: Sure thing.

**Comment by skyrpex** [comment]
This is ready to be merged now! Wing Cloud is ready, and this PR has been tested and checked locally.

**Review by eladb — APPROVED** [review]
🚀

**Pull request merged**
